pkg/common/telemetry/metrics_benchmark_test.go

```diff
@@ -0,0 +1,159 @@
+package telemetry
+
+import (
+	"net"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	key    = []string{"key1", "key2", "key3"}
+	valf   = float32(5.0)
+	labels = []Label{
+		{
+			Name:  "lkey1",
+			Value: "lval1",
+		},
+		{
+			Name:  "lkey2",
+			Value: "lval2",
+		},
+	}
+	valt = time.Now()
+)
+
+func BenchmarkDogStatsd(b *testing.B) {
+	m := getDogStatsdMetricImpl(b)
+
+	benchmarkMetricImpl(b, m)
+}
+
+func BenchmarkInMem(b *testing.B) {
+	m := getInMemMetricImpl(b)
+
+	benchmarkMetricImpl(b, m)
+}
+
+func BenchmarkM3(b *testing.B) {
+	m := getM3MetricImpl(b)
+
+	benchmarkMetricImpl(b, m)
+}
+
+func BenchmarkPrometheus(b *testing.B) {
+	m := getPrometheusMetricImpl(b)
+
+	benchmarkMetricImpl(b, m)
+}
+
+func BenchmarkStatsd(b *testing.B) {
+	listener, err := net.ListenPacket(statsdProtocol, "localhost:")
+	if err != nil {
```

Comment:
nit: `if err != nil` check is not required (performed internally by `require.NoError`)

There are a bunch of similar checks throughout the file.